src/gui/wizard/owncloudsetuppage.cpp

```diff
@@ -210,33 +186,38 @@ bool OwncloudSetupPage::isComplete() const
 
 void OwncloudSetupPage::initializePage()
 {
+    customizeStyle();
+
     WizardCommon::initErrorLabel(_ui.errorLabel);
 
     _authTypeKnown = false;
     _checking = false;
 
     QAbstractButton *nextButton = wizard()->button(QWizard::NextButton);
     auto *pushButton = qobject_cast<QPushButton *>(nextButton);
-    if (pushButton)
+    if (pushButton) {
         pushButton->setDefault(true);
+        // nextButton->setFocus();
+    }
+
+    _ui.leUrl->setFocus();
 
-    // If url is overriden by theme, it's already set and
-    // we just check the server type and switch to second page
-    // immediately.
-    if (Theme::instance()->overrideServerUrl().isEmpty()) {
-        _ui.leUrl->setFocus();
-    } else if (!Theme::instance()->forceOverrideServerUrl()) {
+    if (!Theme::instance()->overrideServerUrl().isEmpty() && !Theme::instance()->forceOverrideServerUrl()) {
+        // If the url is overwritten but we don't force to use that url
+        // Just focus the next button to let the user navigate quicker
         if (nextButton) {
             nextButton->setFocus();
         }
-    } else {
+    } else if (!Theme::instance()->overrideServerUrl().isEmpty()) {
```

Comment:
@FlexW Is it essential for us to check the same condition twice?

Can it be reorganized like the following:

`if (!Theme::instance()->overrideServerUrl().isEmpty()) {  
     if (Theme::instance()->forceOverrideServerUrl()) {
         doA(); 
     } else { 
         doB();
     } 
 }`

Reply:
Should be possible I guess. It would be then like that:
```cpp
    if (!Theme::instance()->overrideServerUrl().isEmpty()) {
        if (!Theme::instance()->forceOverrideServerUrl()) {
            // If the url is overwritten but we don't force to use that url
            // Just focus the next button to let the user navigate quicker
            if (nextButton) {
                nextButton->setFocus();
            }
        } else {
            // If the overwritten url is not empty and we force this overwritten url
            // we just check the server type and switch to next page
            // immediately.
            setCommitPage(true);
            // Hack: setCommitPage() changes caption, but after an error this page could still be visible
            setButtonText(QWizard::CommitButton, tr("&Next >"));
            validatePage();
            setVisible(false);
        }
    }
```
I like the original more because it has less nested branches and comes without the `else` statement. What the original could make a little bit better would be to introduce a local variable like `const auto isServerUrlOverridden = ...;`. Then its more clear that we check almost the same condition.

What do you think?